src/condor_utils/file_transfer.cpp

```diff
@@ -3283,6 +3304,12 @@ FileTransfer::GetTransferAck(Stream *s,bool &success,bool &try_again,int &hold_c
 		error_desc = hold_reason_buf;
 		free(hold_reason_buf);
 	}
+	// If this is the condor_shadow (indicated by IsServer() == true) then update
+	// our file transfer stats to include those sent by the condor_starter
+	ClassAd* transfer_stats = dynamic_cast<ClassAd*>(ad.Lookup("TransferStats"));
+	if (transfer_stats && IsServer()) {
+		Info.stats.CopyFrom(*transfer_stats);
```

Comment:
```suggestion
		Info.stats.Update(*transfer_stats);
```
This should fix the crashes. CopyFrom() basically makes the current ClassAd a clone of the given one, including what ClassAd it's nested inside (if it's nested). Update() just copies all attributes from the given ad into the current one. It doesn't clear out the current ad first, so you may want to add a Clear() as well.